### ISSUE TITLE
aws_ecs_service doesn't provide assignPublicIp option in network_configuration

### ISSUE THREAD

**Issue by maurycyp** [terraform-providers/terraform-provider-aws#2873]
Version: Terraform v0.11.1

Since the `assignPublicIp` option is missing from the `aws_ecs_service` resource, it defaults to DISABLED, which means that a task fails when pulling images from ECR.

You can see the `assignPublicIp` option using the AWS CLI (`aws ecs create-service help`), where it exists alongside the subnet and security group options.

**Comment by Dredalious** [comment]
FYI: A workaround exists using private subnets and NAT gateways. 
But I want the assignPublicIp option as well.

**Comment by JonRoma** [comment]
With Amazon's new support for containerization via Fargate, being able to assign a public IP address is a necessity for some use cases. Accordingly, I would respectfully suggest that this omission is a bug in that a useful AWS feature is not exposed to Terraform users. 

At https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_AWSCLI_Fargate.html, we see the data structure [excerpted]: 

```
"networkConfiguration": {
                "awsvpcConfiguration": {
                    "subnets": [
                        "subnet-abcd1234"
                    ], 
                    "securityGroups": [
                        "sg-abcd1234"
                    ], 
                    "assignPublicIp": "DISABLED"
                }
            }, 
```

In [the AWS provider source](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_ecs_service.go), one can see no mention of the assignPublicIP attribute (in any of its variants).

**Comment by JonRoma** [comment]
I should hasten to add the following detail:
```
: terraform --version
Terraform v0.11.2
```

The AWS provider version is 1.7.

**Comment by austinkelleher** [comment]
Looks like someone opened a PR back in December to fix this issue: https://github.com/terraform-providers/terraform-provider-aws/pull/2559

**Comment by johnnorton** [comment]
#2559 is ready for 🤞 review.  Thanks for your patience.

**Comment by bflad** [comment]
This has been merged into master and will release with v1.9.0 of the AWS provider, hopefully later today.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.9.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#190-february-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
